sync/syncer.go

```diff
@@ -63,6 +69,10 @@ type LayerValidator interface {
 	ValidateLayer(lyr *types.Layer)
 }
 
+type LayerProvider interface {
```

Comment:
This is redundant since Syncer contains mesh.Mesh

Reply:
It is used for testing.
It is very complicated to mock the whole mesh and I didn't want to skip tests/waste a lot of time fixing it all at once.
IMO, this is a step towards having a more testable package. In addition, I would not be afraid of having multiple, smaller, interfaces. I do agree that later on, we shouldn't have the full mesh as a member. Pretty sure there is an issue on that.